### PR TITLE
Add Error Notices to Google My Business Charts

### PR DESCRIPTION
**Pull request by cbauerman**
## Specs

screen shot 2018-05-15 at 3 05 07 pm

Adds notices to charts where an error has occurred while trying to fetch the data.

## Testing

1. Introduce errors to Google My Business APIs ( see patch in D12650 )
2. Navigate to the Google My Business on an eligilbe Stats Page
3. Open browser developer console network tab
4. If necessary, change the interval of the charts until an error notice appears
5. Verify that the responses that have an "error" in the body show an error notice on the chart

## Review
- [ ] Code
- [ ] Product
- [ ] Copy

### CODE REVIEW

File: client/my-sites/google-my-business/stats/chart.js [hunk 1]
```diff
@@ -206,8 +206,34 @@ class GoogleMyBusinessStatsChart extends Component {
...
 
+	renderChartNotice( isError = false ) {
+		const { translate } = this.props;
+
+		const emptyText = translate( 'No activity this period', {
+			context: 'Message on empty bar chart in Stats',
+			comment: 'Should be limited to 32 characters to prevent wrapping',
+		} );
+
+		const errorText = translate( 'Error loading data', {
```
Comment (a8ci18n) [inline client/my-sites/google-my-business/stats/chart.js]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/state/google-my-business/reducer.js [hunk 2]
```diff
@@ -5,13 +5,15 @@
...
 import {
+	GOOGLE_MY_BUSINESS_STATS_FAILURE,
 	GOOGLE_MY_BUSINESS_STATS_RECEIVE,
 	GOOGLE_MY_BUSINESS_STATS_REQUEST,
 } from 'state/action-types';
 
 const stats = createReducer( null, {
 	[ GOOGLE_MY_BUSINESS_STATS_RECEIVE ]: ( state, { data } ) => data,
 	[ GOOGLE_MY_BUSINESS_STATS_REQUEST ]: () => null,
+	[ GOOGLE_MY_BUSINESS_STATS_FAILURE ]: () => false,
```
Comment (Tug) [inline client/state/google-my-business/reducer.js]: I think we usually create an error reducer in those cases.
See https://github.com/Automattic/wp-calypso/blob/master/client/state/login/reducer.js#L108

It's cleaner than setting the data reducer to `false` imo. We can then user a selector to return the error.

File: client/state/data-layer/wpcom/sites/stats/google-my-business/index.js [hunk 3]
```diff
@@ -49,8 +47,20 @@ export const receiveStats = ( { dispatch }, action, data ) => {
 	);
 };
 
+/**
+ * Dispatches a failure to retrieve stats
+ *
+ * @param {Function} dispatch Redux dispatcher
+ * @param {Object} action Redux action
+ */
+export const receiveStatsError = ( { dispatch }, action, error ) => {
```
Comment (yurynix) [inline client/state/data-layer/wpcom/sites/stats/google-my-business/index.js]: Personally I would've opted for a shorter:
```es6
export const receiveStatsError = ( { dispatch }, { siteId, statType, interval, aggregation }, error ) =>
	dispatch( failedRequestGoogleMyBusinessStats( siteId, statType, interval, aggregation, error ) );
```

**Comment by yurynix** [comment]
I've rebased the PR to latest master
The code looks good and works as described 👍 LGTM

I think we should cancel the loading state on an error, but that can be done in a following PR.

File: client/my-sites/google-my-business/stats/chart.js (hunk 1 above)
Comment (akirk) [reply client/my-sites/google-my-business/stats/chart.js]: @cbauerman Could you please convert this to a comment?

**Pull request merged**